src/RecyclableMemoryStream.cs

```diff
@@ -308,13 +311,14 @@ protected override void Dispose(bool disposing)
             }
 
             this.disposed = true;
+            var lifetime = TimeSpan.FromTicks(Stopwatch.GetTimestamp() - this.creationTimestamp);
```

Comment:
This is only correct when the duration of a tick and the timestamp are the same. This is true on Windows now, but it may not be on other platforms or in the future. See the [discussion of why Stopwatch.GetElapsedTime](https://github.com/dotnet/runtime/issues/65858#issuecomment-1050836263) was added for the right way to do it:

```suggestion
            var lifetime = TimeSpan.FromTicks((long)((Stopwatch.GetTimestamp() - this.creationTimestamp) * (TimeSpan.TicksPerSecond / Stopwatch.Frequency)));
```

Reply:
On [Unix](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.Unix.cs,10), for example, the smallest increment of a timestamp is only 1 nanosecond! The time reported would then be 100 times too long.